### PR TITLE
sign gradle plugin

### LINKED ISSUE #192: Plugin not signed
https://docs.gradle.org/6.2/userguide/dependency_verification.html

build.gradle

```
plugins {
  id 'com.github.spotbugs' version '3.0.0'
}

wrapper.configure {
  gradleVersion = '6.2.1'
}
```

Execute:

```
$ ./gradlew --write-verification-metadata pgp,sha256 help
```

`gradle/verification-metadata.xml` will now have the following entries:

```
      <component group="com.github.spotbugs" name="com.github.spotbugs.gradle.plugin" version="3.0.0">
         <artifact name="com.github.spotbugs.gradle.plugin-3.0.0.pom">
            <sha256 value="ad491373febf4bae68c430a44067afcfa901f83a6992533dad5114d8912ba866" origin="Generated by Gradle because artifact wasn't signed"/>
         </artifact>
      </component>
      <component group="com.github.spotbugs" name="spotbugs-gradle-plugin" version="3.0.0">
         <artifact name="spotbugs-gradle-plugin-3.0.0.jar">
            <sha256 value="a5a5d0dd6d9244f74306699bdfe7990af7a504a8db46cc5f2e5cfd48a601e69c" origin="Generated by Gradle because artifact wasn't signed"/>
         </artifact>
         <artifact name="spotbugs-gradle-plugin-3.0.0.pom">
            <sha256 value="d7bfd76800d72fb9151feb53cf34568e6bf0fc31b0d611d66ed009e7f4d30b06" origin="Generated by Gradle because artifact wasn't signed"/>
         </artifact>
      </component>
```

### PR DESCRIPTION
**Pull request by jscancella**
This pull request closes #192 - plugin not signed. Note that some additional work needs to be done by the maintainer, mainly generating the PGP keys, uploading them to the pool servers, and adding the secrets to the github repository. See README for instructions.

### CODE REVIEW

**Comment by jscancella** [comment]
The PR is failing the build because the secrets are not defined. As you can see on mine (https://github.com/jscancella/spotbugs-gradle-plugin/runs/704278787) once they are defined they run fine.

**Comment by jscancella** [comment]
> 
> 
> Thanks for your PR!
> 
> I added two secrets but [GitHub Actions does not use them](https://github.com/spotbugs/spotbugs-gradle-plugin/pull/269/checks?check_run_id=704723626#step:7:7) even though I rerun the workflow.
> 
> ~I guess we need to update the branch (e.g. merging `origin/master`) to refer the latest secrets, could you have a try?~
> 
> It seems that [the encrypted secrets are not accessible for PR from the forked repo](https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#using-encrypted-secrets-in-a-workflow). Then we need to update the Gradle buildscript to pass build without keys.

either that, or merge into a branch and test there to make sure it is working properly.

**Comment by KengoTODA** [comment]
> either that, or merge into a branch and test there to make sure it is working properly.

To confirm it works properly, I created #270. And [it seems working as expected](https://github.com/spotbugs/spotbugs-gradle-plugin/pull/270/checks?check_run_id=705061831).

Please keep working in this PR to ensure that CI build works without secrets, it helps other contributors to submit PRs. :+1:

**Comment by jscancella** [comment]
> 
> 
> > either that, or merge into a branch and test there to make sure it is working properly.
> 
> To confirm it works properly, I created #270. And [it seems working as expected](https://github.com/spotbugs/spotbugs-gradle-plugin/pull/270/checks?check_run_id=705061831).
> 
> Please keep working in this PR to ensure that CI build works without secrets, it helps other contributors to submit PRs. +1

As requested the builds should not fail even if the key/password are not defined for signing the artifacts. What do I need to do to make the `Semantic Pull Request` work correctly?

File: build.gradle [hunk 1]
```diff
@@ -43,6 +44,26 @@ groovydoc {
     link 'https://docs.groovy-lang.org/latest/html/gapi/', 'groovy.', 'org.codehaus.groovy.'
 }
 
+//def signingKey = System.getenv("SIGNING_KEY")
+def signingKey = ""
+//def signingPassword = System.getenv("SIGNING_PASSWORD")
+def signingPassword = ""
```
Comment (KengoTODA) [inline build.gradle:50]: I guess we need to switch back to use `System.getenv`?

Comment (jscancella) [reply build.gradle:50]: Oops, yes. That was from me testing locally that a blank variable was causing the problems. I will update shortly

**Comment by jscancella** [comment]
@KengoTODA why does the one build fail to do the semantic release properly in my fork, but it does not fail here in the pull request?

https://github.com/jscancella/spotbugs-gradle-plugin/runs/710619625?check_suite_focus=true

**Comment by KengoTODA** [comment]
@jscancella according to the log, you have a tag conflict. 

>   stderr: "fatal: tag '3.0.1' already exists",

**Pull request merged**